janusgraph-driver/src/main/java/org/janusgraph/core/attribute/Text.java

```diff
@@ -263,7 +263,9 @@ private static boolean isFuzzy(String term, String value){
         } else {
             distance = 2;
         }
-        return LevenshteinDistance.getDefaultInstance().apply(value, term)<=distance;
+        // LevenshteinDistance returns -1 if distance is larger than threshold
+        int levenshteinDist = new LevenshteinDistance(distance).apply(value, term);
+        return levenshteinDist <= distance && levenshteinDist >= 0;
```

Comment:
Please, don't create a new instance of `LevenshteinDistance` for each search.
We have 3 possible `distance` values (`0`, `1` and `2`). So, just create 3 constants of `LevenshteinDistance` and reuse them.
I.e.
```
private static final LevenshteinDistance ZERO_LEVENSHTEIN_DISTANCE = new LevenshteinDistance(0);
private static final LevenshteinDistance ONE_LEVENSHTEIN_DISTANCE = new LevenshteinDistance(1);
private static final LevenshteinDistance TWO_LEVENSHTEIN_DISTANCE = new LevenshteinDistance(2);
```

After that we can rewrite the method to be as following:
```
private static boolean isFuzzy(String term, String value){
    term = term.trim();
    if (term.length() < 3) {
        return ZERO_LEVENSHTEIN_DISTANCE.apply(value, term) == 0;
    } else if (term.length() < 6) {
        int levenshteinDist = ONE_LEVENSHTEIN_DISTANCE.apply(value, term);
        return levenshteinDist <= 1 && levenshteinDist >= 0;
    }
    int levenshteinDist = TWO_LEVENSHTEIN_DISTANCE.apply(value, term);
    return levenshteinDist <= 2 && levenshteinDist >= 0;
}
```